tests/test_figure_resampler.py

```diff
@@ -426,13 +429,75 @@ def test_multiple_timezones():
+
+
+def multiple_timezones_in_single_x_index():
+    y = np.arange(20)
+    
+    index1 = pd.date_range('2018-01-01', periods=10, freq='H', tz="US/Eastern")
+    index2 = pd.date_range('2018-01-02', periods=10, freq='H', tz="Asia/Dubai")
+    index = index1.append(index2)
+
+    fig = go.Figure()
+    fig.add_trace(go.Scattergl(x=index, y=y))
+    full_fig = fig.full_figure_for_development(warn=False)
+    with pytest.warns(UserWarning):
+        fr_fig = FigureResampler(fig, default_n_shown_samples=10)
+    assert pd.Timestamp(full_fig.layout.xaxis.range[0]) == fr_fig.data[0].x[0]
+    assert pd.Timestamp(full_fig.layout.xaxis.range[1]) == fr_fig.data[0].x[-1]
+    # Add as hf_x as index
+    fr_fig = FigureResampler(default_n_shown_samples=10)
+    with pytest.warns(UserWarning):
+        fr_fig.add_trace(go.Scattergl(), hf_x=index, hf_y=y)
+    assert pd.Timestamp(full_fig.layout.xaxis.range[0]) == fr_fig.data[0].x[0]
+    assert pd.Timestamp(full_fig.layout.xaxis.range[1]) == fr_fig.data[0].x[-1]
+    # Add as hf_x as object array of datetime values
+    fr_fig = FigureResampler(default_n_shown_samples=10)
+    with pytest.warns(UserWarning):
+        fr_fig.add_trace(go.Scattergl(), hf_x=index.values.astype("object"), hf_y=y)
+    assert pd.Timestamp(full_fig.layout.xaxis.range[0]) == fr_fig.data[0].x[0]
+    assert pd.Timestamp(full_fig.layout.xaxis.range[1]) == fr_fig.data[0].x[-1]
+    # Add as hf_x as string array
+    fr_fig = FigureResampler(default_n_shown_samples=10)
+    with pytest.warns(UserWarning):
+        fr_fig.add_trace(go.Scattergl(), hf_x=index.astype(str), hf_y=y)
+    assert pd.Timestamp(full_fig.layout.xaxis.range[0]) == fr_fig.data[0].x[0]
+    assert pd.Timestamp(full_fig.layout.xaxis.range[1]) == fr_fig.data[0].x[-1]
+    # Add as hf_x as object array of strings
+    fr_fig = FigureResampler(default_n_shown_samples=10)
+    with pytest.warns(UserWarning):
+        fr_fig.add_trace(go.Scattergl(), hf_x=index.astype(str).astype("object"), hf_y=y)
+    assert pd.Timestamp(full_fig.layout.xaxis.range[0]) == fr_fig.data[0].x[0]
+    assert pd.Timestamp(full_fig.layout.xaxis.range[1]) == fr_fig.data[0].x[-1]
+
+    fig = go.Figure()
+    fig.add_trace(go.Scattergl(x=index.astype("object"), y=y))
+    full_fig = fig.full_figure_for_development(warn=False)
+    with pytest.warns(UserWarning):
+        fr_fig = FigureResampler(fig, default_n_shown_samples=10)
+    assert pd.Timestamp(full_fig.layout.xaxis.range[0]) == fr_fig.data[0].x[0]
+    assert pd.Timestamp(full_fig.layout.xaxis.range[1]) == fr_fig.data[0].x[-1]
+
+    fig = go.Figure()
+    fig.add_trace(go.Scattergl(x=index.astype("str"), y=y))
+    full_fig = fig.full_figure_for_development(warn=False)
+    with pytest.warns(UserWarning):
+        fr_fig = FigureResampler(fig, default_n_shown_samples=10)
+    assert pd.Timestamp(full_fig.layout.xaxis.range[0]) == fr_fig.data[0].x[0]
+    assert pd.Timestamp(full_fig.layout.xaxis.range[1]) == fr_fig.data[0].x[-1]
```

Comment:
1. Did you forget to start your function name with `test_`? I am not sure this test is launched.

2. Also, I see you are testing the same condition in each example. It might be beneficial to put these asserts into a function, this would make the code more compact, readable and maintainable (just a suggestion, feel free to ignore). Did you know you could also use `pytest.fixture.parametrize` to do the same thing? Though, since you use `.values` and change types multiple times, it may be harder to use for this test...

Reply:
1. Nice catch indeed! This test never got launched
2. Yes I have been thinking this as well.. might perhaps be interesting to refactor the other tests with the `pytest.fixture.parametrize` (but I'll leave that to another PR)